### PR TITLE
feat: deployment to gh-pages on tags

### PR DESCRIPTION
**Pull request by mxschmitt**
also ran sort-imports on the docs directory.

Somebody with permissions e.g. taion or jquense have to set the `GITHUB_TOKEN` environment on travis: https://travis-ci.org/react-bootstrap/react-bootstrap/settings

Reference for the GitHub token: https://docs.travis-ci.com/user/deployment/pages/#setting-the-github-token

### CODE REVIEW

**Comment by taion** [comment]
is there a benefit to having just the GH pages push be from travis here? it seems a little of an odd halfway thing?

**Comment by mxschmitt** [comment]
> is there a benefit to having just the GH pages push be from travis here? it seems a little of an odd halfway thing?

What do you mean with just the GH pages? Maybe we can say on Netlify side, that it will serve the docs of the latest release. Then we can omit this travis-deployment thing and redirect GH-pages to the latest release Netlify docs. This PR just wants to automate the deployment process and fix (wrote jquense on Discord) the codecov service.

**Comment by bpas247** [comment]
@mxschmitt do you mind if I push a commit to resolve the merge conflict? I wanna get this PR going again :)

**Comment by mxschmitt** [comment]
> @mxschmitt do you mind if I push a commit to resolve the merge conflict? I wanna get this PR going again :)

sure if you want. :)

**Comment by bpas247** [comment]
Hmm, what’s causing TravisCI to hang? I don’t even see the build showing up in the logs...

**Comment by taion** [comment]
probably random flake. push another commit to the branch?

**Comment by taion** [comment]
actually, the .travis.yml might be invalid – check the indenting on the `after_success` block?

File: .travis.yml [hunk 1]
```diff
@@ -29,9 +29,20 @@ install:
   - yarn bootstrap
 
 after_success:
-  - node_modules/.bin/codecov
+  - if ([ ! -z "$TRAVIS_TAG" ]); then
```
Comment (taion) [inline .travis.yml]: i think you want https://yaml-multiline.info here with the "keep newlines" syntax (just `|`)

**Comment by mxschmitt** [comment]
Closed due we will implement it via GitHub Actions.

**Pull request closed without merging**